src/BinSkim.Rules/PERules/WindowsBinaryAndPdbSkimmerBase.cs

```diff
@@ -130,20 +134,27 @@ public static void LogExceptionLoadingPdb(IAnalysisContext context, PdbException
                 throw new ArgumentNullException(nameof(context));
             }
 
-            // '{0}' was not evaluated for check '{1}' because its PDB could not be loaded ({2}).
+            string fileName = context.TargetUri.GetFileName();
+            string key = $"{fileName}@{pdbException.ExceptionDisplayMessage}";
+            if (s_PdbExceptions.ContainsKey(key))
+            {
+                return;
+            }
+
+            // '{0}' was not evaluated because its PDB could not be loaded ({1}).
             context.Logger.LogConfigurationNotification(
                 Errors.CreateNotification(
                     context.TargetUri,
                     "ERR997.ExceptionLoadingPdb",
-                    context.Rule.Id,
+                    string.Empty,
                     FailureLevel.Error,
                     pdbException,
                     persistExceptionStack: false,
-                    RuleResources.ERR997_ExceptionLoadingPdb,
-                    context.TargetUri.GetFileName(),
-                    context.Rule.Name,
+                    RuleResources.ERR997_ExceptionLoadingPdbGeneric,
+                    fileName,
                     pdbException.ExceptionDisplayMessage));
 
+            s_PdbExceptions.TryAdd(key, true);
```

Comment:
>s_PdbExceptions.TryAdd [](http://example.com/codeflow?start=12&length=22)

Once we log the notification, we will add to the caching variable. Since it's static and concurrent, it won't face any issue during analysis